File: returned/python/today8/validate_binary_search_tree_test.py
from validate_binary_search_tree import Solution, TreeNode

def test1():
    a = TreeNode(1)
    b = TreeNode(2)
    c = TreeNode(3)
    b.left = a
    b.right = c
    assert Solution().isValidBST(b) == True

def test2():
    n5 = TreeNode(5)
    n4 = TreeNode(4)
    n6 = TreeNode(6)
    n3 = TreeNode(3)
    n7 = TreeNode(7)
    n5.left = n4
    n5.right = n6
    n6.left = n3
    n6.right = n7

    assert Solution().allGreaterThan(n5, n6) == False
    assert Solution().isValidBST(n5) == False

File: csharp/LeetTests/Today4/ValidateBinarySearchTreeTest.cs
using Today4.ValidateBinarySearchTree;
namespace Today4;

internal class ValidateBinarySearchTreeTest()
{
    [Test]
    public void Test1()
    {
        var solution = new Solution();
        var n2 = new TreeNode(2);
        var n1 = new TreeNode(1);
        var n3 = new TreeNode(3);
        n2.left = n1;
        n2.right = n3;
        var result = solution.IsValidBST(n2);
        Assert.That(result, Is.EqualTo(true));
    }

    [Test]
    public void Test2()
    {
        var solution = new Solution();
        var n5 = new TreeNode(5);
        var n1 = new TreeNode(1);
        var n4 = new TreeNode(4);
        var n3 = new TreeNode(3);
        var n6 = new TreeNode(6);
        n5.left = n1;
        n5.right = n4;
        n4.left = n3;
        n4.right = n6;
        var result = solution.IsValidBST(n5);
        Assert.That(result, Is.EqualTo(false));
    }

    [Test]
    public void Test3()
    {
        var solution = new Solution();
        var n5 = new TreeNode(5);
        var n4 = new TreeNode(4);
        var n6 = new TreeNode(6);
        var n3 = new TreeNode(3);
        var n7 = new TreeNode(7);
        n5.left = n4;
        n5.right = n6;
        n6.left = n3;
        n6.right = n7;
        var result = solution.IsValidBST(n5);
        Assert.That(result, Is.EqualTo(false));
    }

    [Test]
    public void Test4()
    {
        var solution = new Solution();
        var n = new TreeNode(2_147_483_647);
        var result = solution.IsValidBST(n);
        Assert.That(result, Is.EqualTo(true));
    }
}